### DISCUSSION TITLE
n8n via ttek script behind nginx reverse proxy - env not albe to be set

### DISCUSSION THREAD

**Discussion by rewulff** [tteck/Proxmox#2418] [Help Needed]
did anyone deployed the n8n lxc via the script and was able to run int behind an nginx rp?

if I try to setup oauth credentials my Redirect URL  is always http://localhost:5678 ...

I tryed to use "export" like mentioned in the n8n setup guid.

Am I missing the elephant in the room?

Unlike the normal npm installation n8n is started as a service.

can anyone ad a guide how to setup env please?

**Comment by rewulff** [comment]
[solution](https://github.com/tteck/Proxmox/issues/1600#issuecomment-1647006540)

**Comment by rewulff** [comment]
The problem continues when installing for example pupeteer via npm. 
The n8n.service won't find the executable path without fiddling.

My honest suggestion is to rework the install script to match the npm standard +np2 service.

atm it seems to be better to install n8n manually on a clea lxc or debian 12 vm.

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/blob/84b91141ff1433443892b4523f7b476540c13125/install/n8n-install.sh#L35-L37

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/blob/84b91141ff1433443892b4523f7b476540c13125/install/n8n-install.sh#L40-L49

**Reply by tteck** [reply]
Have you attempted to set your Environment variable in the service file?

**Reply by rewulff** [reply]
Thx for your reply.

This is the workaround for using it as a service. (found after a whole day digging the rabit hole. 
While this is off the n8n guide, the bigger problem is to use plugins installed by npm. 

for example npm i puppeteer wont be reachable for the n8n instance runing as a service.

the n8n core runs flawles (after adding ev to service file), but addons have issues when used as as system service.

futhermore there is a simple alternative, well documented by n8n community.
https://blog.n8n.io/how-to-set-up-n8n-via-pm2/

**Comment by tteck** [comment]
Does the n8n application run smoothly without any issues when not using your hacks?

**Reply by tteck** [reply]
Why would anyone run n8n behind a proxy? What are you attempting to accomplish?

**Reply by rewulff** [reply]
The n8n itself runs smoothly.

I wanted to use OAuth authenticated connections, e.g., Gmail and MS365. To achieve that, I have to point to an OAuth Redirect URL on my n8n instance (configured by env).

To have this in a secure way, I am using a firewall, IDS, IPS, and reverse proxy. (However, this makes no difference at all.) It is working fine and makes me feel much more comfortable instead of opening up the ports transparently to the internet.

Next, I am trying to automate PDF converting by Puppeteer or at least Gotenberg locally (installed by npm on the shell). That's the step I am stuck right now because n8n is running as a system service. The only way to go without fiddling is to not run it as a system service.

I tried running it directly off the shell by stopping the service and starting it using npm start. In that case, everything is working as expected, and changes can be made as documented in the official n8n FAQ.

**Comment by tteck** [comment]
/
